tests/e2e/Services/VCS/VCSConsoleClientTest.php

```diff
@@ -0,0 +1,316 @@
+
+use Tests\E2E\Scopes\Scope;
+use Tests\E2E\Client;
+use Tests\E2E\Scopes\ProjectCustom;
+use Tests\E2E\Scopes\SideConsole;
+use Utopia\App;
+use Utopia\Database\Helpers\ID;
+use Utopia\Database\Helpers\Role;
+use Utopia\VCS\Adapter\Git\GitHub;
+use Utopia\Cache\Adapter\None;
+use Utopia\Cache\Cache;
+
+class VCSConsoleClientTest extends Scope
+{
+    use ProjectCustom;
+    use SideConsole;
+
+    public ?string $providerRepositoryId = null;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->providerRepositoryId = '705764267';
+    }
+
+    public function testGitHubAuthorize()
+    {
+        /**
+         * Test for SUCCESS
+         */
+        $response = $this->client->call(Client::METHOD_GET, '/mock/github/callback', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), [
+            'installation_id' => App::getEnv('_APP_VCS_TEST_GITHUB_INSTALLATION_ID'),
+            'projectId' => $this->getProject()['$id'],
+        ]);
+
+        $this->assertNotEmpty($response['body']['installationId']);
+        $installationId = $response['body']['installationId'];
+        return $installationId;
+    }
+
+    /**
+     * @depends testGitHubAuthorize
+     */
+    public function testGetInstallation(string $installationId)
+    {
+        /**
+         * Test for SUCCESS
+         */
+
+        $installation = $this->client->call(Client::METHOD_GET, '/vcs/installations/' . $installationId, array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), [
+            'installationId' => $installationId
+        ]);
+
+        $this->assertEquals(200, $installation['headers']['status-code']);
+    }
+
+    /**
+     * @depends testGitHubAuthorize
+     */
+    public function testDetectRuntime(string $installationId)
+    {
+        /**
+         * Test for SUCCESS
+         */
+
+        $runtime = $this->client->call(Client::METHOD_POST, '/vcs/github/installations/' . $installationId . '/providerRepositories/' . $this->providerRepositoryId . '/detection', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), [
+            'installationId' => $installationId,
+            'providerRepositoryId' => $this->providerRepositoryId
+        ]);
+
+        $this->assertEquals(200, $runtime['headers']['status-code']);
+        $this->assertEquals($runtime['body']['runtime'], 'ruby-3.1');
+
+        /**
+         * Test for FAILURE
+         */
+
+        $runtime = $this->client->call(Client::METHOD_POST, '/vcs/github/installations/' . $installationId . '/providerRepositories/randomRepositoryId/detection', array_merge([
+            'content-type' => 'application/json',
```

Comment:
Content type should not be needed

Reply:
Its in multiple places of this file. See if it works without it. If yes, lets remove it.

Reply:
Removed it from wherever the tests were working without it.